File: app/src/main/java/me/wcy/music/utils/ScreenUtils.java
package me.wcy.music.utils;

import android.content.Context;
import android.text.TextUtils;
import android.view.WindowManager;

import me.wcy.music.application.MusicApplication;

/**
 * 工具类
 * Created by hzwangchenyan on 2016/1/6.
 */
public class ScreenUtils {
    public static int getScreenWidth() {
        WindowManager wm = (WindowManager) MusicApplication.getInstance().getSystemService(Context.WINDOW_SERVICE);
        return wm.getDefaultDisplay().getWidth();
    }

    /**
     * 获取状态栏高度
     */
    public static int getSystemBarHeight(Context context) {
        int result = 0;
        int resourceId = context.getResources().getIdentifier("status_bar_height", "dimen", "android");
        if (resourceId > 0) {
            result = context.getResources().getDimensionPixelSize(resourceId);
        }
        return result;
    }

    public static String getArtistAndAlbum(String artist, String album) {
        if (TextUtils.isEmpty(artist) && TextUtils.isEmpty(album)) {
            return "";
        } else if (!TextUtils.isEmpty(artist) && TextUtils.isEmpty(album)) {
            return artist;
        } else if (TextUtils.isEmpty(artist) && !TextUtils.isEmpty(album)) {
            return album;
        } else {
            return artist + " - " + album;
        }
    }

    public static int dp2px(float dpValue) {
        Context context = MusicApplication.getInstance().getApplicationContext();
        final float scale = context.getResources().getDisplayMetrics().density;
        return (int) (dpValue * scale + 0.5f);
    }

    public static int px2dp(float pxValue) {
        Context context = MusicApplication.getInstance().getApplicationContext();
        final float scale = context.getResources().getDisplayMetrics().density;
        return (int) (pxValue / scale + 0.5f);
    }

    public static int sp2px(float spValue) {
        Context context = MusicApplication.getInstance().getApplicationContext();
        final float fontScale = context.getResources().getDisplayMetrics().scaledDensity;
        return (int) (spValue * fontScale + 0.5f);
    }
}

File: app/src/main/java/me/wcy/music/utils/ScreenUtils.kt
package me.wcy.music.utils

import android.content.Context
import android.view.WindowManager

/**
 * 工具类
 * Created by hzwangchenyan on 2016/1/6.
 */
object ScreenUtils {
    private var sContext: Context? = null
    fun init(context: Context?) {
        sContext = context!!.applicationContext
    }

    val screenWidth: Int
        get() {
            val wm = sContext!!.getSystemService(Context.WINDOW_SERVICE) as WindowManager
            return wm.defaultDisplay.width
        }
    val statusBarHeight: Int
        /**
         * 获取状态栏高度
         */
        get() {
            var result = 0
            val resourceId =
                sContext!!.resources.getIdentifier("status_bar_height", "dimen", "android")
            if (resourceId > 0) {
                result = sContext!!.resources.getDimensionPixelSize(resourceId)
            }
            return result
        }

    fun dp2px(dpValue: Float): Int {
        val scale = sContext!!.resources.displayMetrics.density
        return (dpValue * scale + 0.5f).toInt()
    }

    fun px2dp(pxValue: Float): Int {
        val scale = sContext!!.resources.displayMetrics.density
        return (pxValue / scale + 0.5f).toInt()
    }

    fun sp2px(spValue: Float): Int {
        val fontScale = sContext!!.resources.displayMetrics.scaledDensity
        return (spValue * fontScale + 0.5f).toInt()
    }
}